build/codegen/codegen.sh

```diff
@@ -14,29 +14,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-GROUP_VERSIONS="ceph.rook.io:v1"
+set -o pipefail
+set -o nounset
+set -o errexit
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
-scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+base_dir="$(cd "${script_dir}/../.."  && pwd)"
+
+boilerplate="${base_dir}"/build/codegen/header.txt
+source "${CODE_GENERATOR}/kube_codegen.sh"
```

Comment:
recording the location of the boilerplate header file is not related to loading the codegen file. Separate these lines.

```suggestion
boilerplate="${base_dir}"/build/codegen/header.txt

source "${CODE_GENERATOR}/kube_codegen.sh"
```